miio/cooker_multi.py

```diff
@@ -0,0 +1,419 @@
+import enum
+import logging
+import math
+from collections import defaultdict
+from typing import List, Optional
+
+import click
+import crcmod
+
+from .click_common import command, format_output
+from .device import Device, DeviceStatus
+from .exceptions import DeviceException
+
+_LOGGER = logging.getLogger(__name__)
+
+MODEL_MULTI = "chunmi.cooker.eh1"
+
+COOKING_STAGES = {
+    1: {
+        "name": "Quickly preheat",
+        "description": "Increase temperature in a controlled manner to soften rice",
+    },
+    2: {
+        "name": "Absorb water at moderate temp.",
+        "description": "Increase temperature steadily and let rice absorb enough water to provide full grains and a taste of fragrance and sweetness.",
+    },
+    3: {
+        "name": "Operate at full load to boil rice",
+        "description": "Keep heating at high temperature. Let rice to receive thermal energy uniformly.",
+    },
+    4: {
+        "name": "Operate at full load to boil rice",
+        "description": "Keep heating at high temperature. Let rice to receive thermal energy uniformly.",
+    },
+    5: {
+        "name": "Operate at full load to boil rice",
+        "description": "Keep heating at high temperature. Let rice to receive thermal energy uniformly.",
+    },
+    6: {
+        "name": "Operate at full load to boil rice",
+        "description": "Keep heating at high temperature. Let rice to receive thermal energy uniformly.",
+    },
+    7: {
+        "name": "Ultra high",
+        "description": "High-temperature steam generates crystal clear rice grains and saves its original sweet taste.",
+    },
+    9: {
+        "name": "Cook rice over a slow fire",
+        "description": "Keep rice warm uniformly to lock lateral heat inside. So the rice will get gelatinized sufficiently.",
+    },
+    10: {
+        "name": "Cook rice over a slow fire",
+        "description": "Keep rice warm uniformly to lock lateral heat inside. So the rice will get gelatinized sufficiently.",
+    },
+}
+
+COOKING_MENUS = {
+    "0000000000000000000000000000000000000001": "Fine Rice",
+    "0101000000000000000000000000000000000002": "Quick Rice",
+    "0202000000000000000000000000000000000003": "Congee",
+    "0303000000000000000000000000000000000004": "Keep warm",
+}
+
+
+class CookerException(DeviceException):
+    pass
+
+
+class OperationMode(enum.Enum):
+    Waiting = 1
+    Running = 2
+    AutoKeepWarm = 3
+    PreCook = 4
+
+    Unknown = "unknown"
+
+    @classmethod
+    def _missing_(cls, _):
```

Comment:
I don't think this works, should be `__missing__`?

Reply:
I am not an python expert. But from this doc it looks good to me 
[https://docs.python.org/3/library/enum.html#enum.Enum._missing_](https://docs.python.org/3/library/enum.html#enum.Enum._missing_)